File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        n=len(nums);
        if(n == 0):
            return 0
        i=1
        ml=1
        st=set(nums)
        for num in nums:
            if(num-1 in st):
                continue
            else:
                c=1
                while(num+c in st):
                    c+=1;
            ml = max(ml,c)
        return ml
        
        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n=nums.size();
        if(n==0) return 0;
        unordered_set<int> st;
        int ml=1;
        for(int i=0;i<n;i++) st.insert(nums[i]);
        for(int i=0;i<n;i++){
            int c=1;
            if(st.find(nums[i]-1)==st.end()){
                while(st.find(nums[i]+c)!=st.end()) c++;
            }
            ml=max(c,ml);
        }
        return ml;
    }
};